src/Main.elm

```diff
@@ -2,72 +2,137 @@ module Main exposing (main)
+    , Cmd.none
+    )
+    
+    
+    
 ---- UPDATE ----
 
 
 type Msg
     = NoOp
+    | ContactDetailsMsg ContactDetails.Msg
+    | SettingsMsg Settings.Msg
+    | TagsMsg Tags.Msg
 
 
 update : Msg -> Model -> ( Model, Cmd Msg )
-update _ model =
-    ( model, Cmd.none )
+update msg model =
+    case model of
+        Ok state ->
+            case msg of
+                NoOp ->
+                    ( model, Cmd.none )
+                
+                ContactDetailsMsg contactDetailsMsg ->
+                    let
+                        ( newState, cmd ) =
+                            ContactDetails.update contactDetailsMsg state.contactDetails
+                    in
+                        ( Ok { state | contactDetails = newState }
+                        , Cmd.map ContactDetailsMsg cmd
+                        )
+
+                SettingsMsg settingsMsg ->
+                    let
+                        ( newState, cmd ) =
+                            Settings.update settingsMsg state.settings
+                    in
+                        ( Ok { state | settings = newState }
+                        , Cmd.map SettingsMsg cmd
+                        )
+
+                TagsMsg tagsMsg ->
+                    let
+                        ( newState, cmd ) =
+                            Tags.update tagsMsg state.tags
+                    in
+                        ( Ok { state | tags = newState }
+                        , Cmd.map TagsMsg cmd
+                        )
+
+        Err _ ->
+            ( model, Cmd.none )
 
 
 
 ---- VIEW ----
 
 
-header : String -> Html msg
-header text =
-    Html.span [ Attrs.class "p-2 text-5xl font-extrabold text-transparent bg-clip-text bg-gradient-to-br from-slate-400 to-slate-800" ]
-        [ Html.text text ]
-
-
-subheader : String -> Html msg
-subheader text =
-    Html.span [ Attrs.class "p-2 text-2xl font-extrabold text-slate-800" ]
-        [ Html.text text ]
-
-
 view : Model -> Html Msg
-view _ =
-    Html.div [ Attrs.class "flex flex-col w-[1024px] items-center mx-auto mt-16 mb-48" ]
-        [ header "Let's start your task"
-        , subheader "Here are your data:"
-        , Html.pre [ Attrs.class "my-8 py-4 px-12 text-sm bg-slate-100 font-mono shadow rounded" ] [ Html.text Data.userGroup ]
-        , header "Now turn them into form."
-        , subheader "See README for details of the task. Good luck 🍀 "
+view model =
+    Html.div [ Attrs.class "mx-auto max-w-7xl px-4 py-24 sm:px-6 sm:py-32 lg:px-8" ]
+        [ case model of
+              Ok { name, contactDetails, settings, tags } ->
+                  Html.div []
+                      [ headerView name
+                      , ContactDetails.view contactDetails
+                      |> Html.map ContactDetailsMsg
+                      , Settings.view settings
+                      |> Html.map SettingsMsg
+                      , Tags.view tags
+                      |> Html.map TagsMsg
+                      ]
```

Comment:
Q: having `|>` aligned with `,` seems odd to me. Do you use altered elm-format? What are the benefits of using custom formatter to you?

Reply:
I use Emacs and this behaviour is the default of the elm-mode in Emacs for some reason. I'm annoyed by it as well to be honest, because it looks ugly. Maybe I should have used elm-format before handing in the test. Didn't think about it.

Reply:
I thought it would be something like that.